### DISCUSSION TITLE
Python 3 universal2 vs. Intel

### DISCUSSION THREAD

**Discussion by gmcluhan** [pyenv/pyenv#2119] [Q&A] [answered]
Hi there I was wondering when you run `pyenv install 3.9.7` will it install the universal2 or intel version of Python?

https://www.python.org/downloads/release/python-39

<img width="778" alt="Screen Shot 2021-10-21 at 10 44 15 PM" src="https://user-images.githubusercontent.com/87726340/138399476-43a84b76-7d75-46fe-ab1f-bbdc1a6ba114.png">

**Comment by native-api** [comment, marked as answer]
We build for the current architecture by default. To get a universal binary, put `--enable-universalsdk` into the `PYTHON_CONFIGURE_OPTS` envvar.

To be more specific, Pyenv doesn't use either of the above installers. It rather downloads the source code and builds Python from it.

**Reply by tifrueh** [reply]
Should this work on M1, too? I'm just asking because the build fails if I add `--enable-universalsdk` to the variable.

**Reply by native-api** [reply]
Report the problem in an issue with the required diagnostic info for us to be able to say anything about it.

**Reply by tifrueh** [reply]
Okay, I'm going to do that right away :)

**Reply by gmcluhan** [reply]
What ended up for working for me on M1 was: `PYTHON_CONFIGURE_OPTS="--with-universal-archs" pyenv install 3.9.7` When I did `--enable-universalsdk` there was an error in the logs that suggested `--with-universal-archs` and that worked.

**Reply by tifrueh** [reply]
That's right, the installation goes through like this. But the binary doesn't actually end up being a universal one, but an arm64 one. Or at least when I tried it.

**Reply by gmcluhan** [reply]
I just updated with the PR they merged for this. How do you actually tell if the version it installed in the universal2 version?

**Reply by tifrueh** [reply]
Use the command `file` on the executable. It will tell you which architectures the contained binaries are for.
